### ISSUE TITLE
Container keeps restarting every 30s or less (v4.0_armhf on a 1st-gen RasPi B+)

### ISSUE THREAD

**Issue by rlue** [pi-hole/docker-pi-hole#336]
This is a...

- [ ] Request for a new or modified feature
- [x] Issue trying to run the docker image
- [ ] Issue trying to build / test / develop the docker image

## Description

Ran a new container with the following command (adapted from `docker-run.sh`):

```
$ docker run -d --name pihole \
  -p 53:53/tcp -p 53:53/udp \
  -p 67:67/udp \
  -p 80:80 \
  -p 443:443 \
  -v "$HOME/pihole/pihole/:/etc/pihole/" \
  -v "$HOME/pihole/dnsmasq.d/:/etc/dnsmasq.d/" \
  -e ServerIP=192.168.1.123 \
  --restart=unless-stopped \
  pihole/pihole:v4.0_armhf
```

## Expected Behavior

Container should boot once and stay on.

## Actual Behavior

Container continually restarts. Can't tell precisely how long it stays on for before restarting, because the system is currently under load and even running `docker container ls` takes ~30s to complete. But so far, I haven't observed it to be running for more than 33s at a time.

Output of `docker log` is a loop of the following:

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 01-resolver-resolv: applying...
[fix-attrs.d] 01-resolver-resolv: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 20-start.sh: executing...
[cont-init.d] 20-start.sh: exited 267.
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] syncing disks.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
```

Using `pihole/pihole:latest` results in the following log message instead:

```
standard_init_linux.go:190: exec user process caused "exec format error"
```

## Debug steps I have tried

- [x] I have tried destroying my container instance, pulling the newest image version, and re-creating a new container
- [x] I have tried running the nearly stock `docker run` example in the readme (removing any customizations I added)
- [x] I have tried running without my volume data mounts to eliminate volumes as the cause
- [x] I have searched this repository for existing issues and pull requests that look similar 

## Your Environment 

* Docker Host Operating System and OS Version: Raspbian 9.4 (stretch)
* Docker Version: 18.05.0-ce, build f150324
* Hardware architecture: ARMv6 (1st-gen Raspberry Pi Model B Plus Rev 1.2)

**Comment by monouser7dig** [comment]
same here

**Comment by diginc** [comment]
The original RBPi armv6 docker install seems to have issues running armhf containers, even though it can run armhf binaries just fine.  Sorry, no solutions or tricks to work around this besides just using the standard non-docker pi-hole install.

**Comment by monouser7dig** [comment]
It works ok on another pi (using docker pihole), not sure yet what’s the difference, the other one is B+ but should not matter I guess.

**Comment by reloxx13** [comment]
same here, pi3b+ and dev tag

`standard_init_linux.go:190: exec user process caused "no such file or directory"`

**Comment by rlue** [comment]
fwiw, I think the `dev` tag is for x64 architecture, not arm. (Correct me if I'm wrong @diginc)

**Comment by diginc** [comment]
latest and dev tag now use magic to auto detect the architecture now @rlue , no more having to specify the arch in the image name/tag thanks to recent enhancements.  

I duplicated @reloxx13 's problem with the dev tag on my RBPi 2.  I don't want to confuse people into think RBPi 1/0s should NOT have this architecture error message (`standard_init_linux.go:190: exec user process caused "no such file or directory"`) by continuing the discussion here, so I'll create a new issue for the dev tag error on RBPi2+ hardware.

Again, `standard_init_linux.go:190: exec user process caused "no such file or directory"` is expected if your `uname -m` says **armv6l**.  If you have **armv7l** or greater newer than 7l, then you should be able to run pihole docker.

**Comment by diginc** [comment]
`exec user process caused "exec format error"` error has been resolved

**Comment by DanTup** [comment]
FWIW, I had a different problem that resulted in infinite reboots that I just tracked down that may be useful to note here for anyone searching for solutions. I'd accidentally added static DHCP IP for one that was already taken - there was no error but the next time I rebooted it got stuck in a loop.

I figured it out by running `docker logs pihole` as it was booting, and saw this in the output:

dnsmasq: duplicate dhcp-host IP address 192.168.0.116 at line 25 of /etc/dnsmasq.d/04-pihole-static-dhcp.conf

I was able to fix by just manually editing that file from my docker volume folder.
